core-java-modules/core-java-14/src/main/java/com/baeldung/java14/recordsCustomConstructors/Main.java

```diff
@@ -0,0 +1,34 @@
+package org.example;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class Main {
+    public static void main(String[] args) {
+        StudentClassic sc = new StudentClassic("Hajar", 2, 18);
+        StudentRecord s= new StudentRecord("Hajar", 2, 18);
+        StudentRecord s1= new StudentRecord("Meta");
+        System.out.println(s.name());
+        System.out.println(s.equals(sc));
+        System.out.println(sc.getName());
+
+        List<StudentRecord> studentRecords = List.of(
+                new StudentRecord("Dana", 1, 85),
+                new StudentRecord("Jim", 2, 90),
+                new StudentRecord("Jane", 3, 80)
+        );
+
+        List<StudentRecord> mutableStudentRecords = new ArrayList<>(studentRecords);
+        mutableStudentRecords.sort(Comparator.comparing(StudentRecord::name));
+
+        List<StudentRecord> sortedStudentRecords = List.copyOf(mutableStudentRecords);
+        System.out.println(sortedStudentRecords);
+
+        for(StudentRecord ss: sortedStudentRecords){
+            System.out.println(ss);
+        }
+
+        StudentRecordV2 srV2 = new StudentRecordV2("Hajar", 4, 80);
+        System.out.println(srV2.grade());
+    }
```

Comment:
Would it make more sense to move what we're trying to show here into individual tests?

We don't really need a main method for these examples, so it might be clearer without this, or to have it moved into individual unit tests where what we're showing can be well-explained in the context of the test.